tests/test_api_machines_serviceid.py

```diff
@@ -0,0 +1,310 @@
+            self.assertEqual(value[1]["application"], "ssh")
+            self.assertEqual(value[1].get("options").get("service_id"), self.serviceID2)
+            self.assertEqual(value[1].get("serial"), self.serial2)
+
+        # Get token for service_id self.serviceID2 and the wrong application
+        with self.app.test_request_context(
+                '/machine/token?service_id={0!s}&application=openssh'.format(self.serviceID2),
+                method='GET',
+                headers={'Authorization': self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 200, res)
+            result = res.json.get("result")
+            self.assertEqual(result["status"], True)
+            value = result.get("value")
+            self.assertEqual(len(value), 0)
+
+    def test_04_get_service_id_for_different_users(self):
+        # Add another SSH key for user admin to self.serviceID2
+        with self.app.test_request_context('/machine/token',
+                                           method='POST',
+                                           data={"serial": self.serial1,
+                                                 "application": "ssh",
+                                                 "user": "admin",
+                                                 "service_id": self.serviceID2},
+                                           headers={'Authorization': self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 200, res)
+            result = res.json.get("result")
+            self.assertEqual(result["status"], True)
+            self.assertTrue(result["value"] >= 1)
+            mtid = result.get("value")
+
+        # Get token for service_id self.serviceID2 and the application=ssh and the user=root
+        with self.app.test_request_context(
+                '/machine/token?service_id={0!s}&application=ssh&user=root'.format(self.serviceID2),
+                method='GET',
+                headers={'Authorization': self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 200, res)
+            result = res.json.get("result")
+            self.assertEqual(result["status"], True)
+            value = result.get("value")
+            self.assertEqual(len(value), 2)
+            self.assertEqual(value[0]["application"], "ssh")
+            self.assertEqual(value[0].get("options").get("service_id"), self.serviceID2)
+            self.assertEqual(value[0].get("serial"), self.serial1)
+            self.assertEqual(value[1]["application"], "ssh")
+            self.assertEqual(value[1].get("options").get("service_id"), self.serviceID2)
+            self.assertEqual(value[1].get("serial"), self.serial2)
+
+        with self.app.test_request_context(
+                '/machine/token?service_id={0!s}&application=ssh&user=admin'.format(self.serviceID2),
```

Comment:
"# Get token for service_id self.serviceID2 and the application=ssh and the user=**admin**"

Reply:
Done in f4469ac